### ISSUE TITLE
Does not load: rubygems.rb

### ISSUE THREAD

**Issue by XULRunner42** [shoes/shoes-deprecated#222]
I was able to get shoes downloaded and compiled by rake

I am using rvm head with ruby 1.9.3-p194, the latest as of this writing
The rake succeeds, producing a couple of files in dist/ and not complaining about anything
(I did not get rake to succeed on arm platform unfortunately because of errors I could not get around related to libgif/libungif)

However, the shoes gem produced claims it does not do anything, and dist/shoes or $SHOES_DIR/dist/shoes produce this error:

```
yebyen@oneiric64:~$ $SHOES_DIR/dist/shoes
<internal:gem_prelude>:1:in `require': cannot load such file -- rubygems.rb (LoadError)
from <internal:gem_prelude>:1:in `<compiled>'
uninitialized constant Encoding::UTF_7
```

I think it should be obvious, I have gems installed, I did gem install bundler and bundle install prior to rake.  For now I am using green shoes, which seems to work OK from the first inspection.

Supposing I should mention, I followed the instructions on https://github.com/shoes/shoes/wiki/Building-Shoes-on-Linux
I am using Ubuntu Precise with an up-to-date rvm ruby, I did not install ruby1.9.1-full as is suggested at the top of the wiki.

Thanks for looking at this!

**Comment by steveklabnik** [comment]
> I did not get rake to succeed on arm platform

I know @nategood knows something about this...

**Comment by nategood** [comment]
I did get it to build on an ARM architecture (Raspberry Pi) and have the build script saved on another box.  I'll dig it up and post it as a gist shortly.

**Comment by nanounanue** [comment]
Hi

I have the same problem that @XULRunner42, I succesfully follow the instructions in  https://github.com/shoes/shoes/wiki/Building-Shoes-on-Linux, except the `ruby1.9.1-full` part, since I am also using  `rvm` without problems.

I am using Linux Mint and Debian (In both distros got the same error)

But when I try to run `{SHOES_HOME}\dist\shoes` I got the following error:

```
<internal:gem_prelude>:1:in `require': cannot load such file -- rubygems.rb (LoadError)
    from <internal:gem_prelude>:1:in `<compiled>'
uninitialized constant Encoding::UTF_7
```

I really would prefer use Shoes, than green shoes, Could you help me?

Thanks in advance

**Comment by PragTob** [comment]
Hm have you been following the debian instructions? What does your ruby -v say?

I run Linux Mint Debian Edition (so basically the combination of your 2 systems ;-)) and it worked like a charm.

Cheers,
Tobi

**Comment by nanounanue** [comment]
Hi Tobi (@PragTob), thanks for your quick response

The oitput of `ruby -v` is the following:

```
nano:~/proyectos/shoes [git:develop?]  
→ ruby -v
ruby 1.9.3p286 (2012-10-12 revision 37165) [x86_64-linux]
```

Cheers and thanks again

Adolfo

**Comment by nategood** [comment]
Okay.  Sorry this took so long.  Here is the build script that worked on my Pi running Raspbian (sorry don't have the Pi with me, don't know the version) with a 8GB class 10 SD.

``` shell
sudo apt-get install git libcairo2-dev libpixman-1-dev libpango1.0-dev libungif4-dev libjpeg62-dev libgtk2.0-dev libsqlite3-dev libcurl4-openssl-dev portaudio19-dev ruby1.9.1-full rake
git clone git://github.com/shoes/shoes.git
git clone git://github.com/hacketyhack/hacketyhack.git
gem install bundler
cd shoes
gem install nokogiri -v '1.5.5'
bundle install
rake
ln -s dist/shoes /usr/bin/shoes
```

**Comment by PragTob** [comment]
Thaks for posting this, I really can't replicate the problems... just recompiled shoes just to make sure and everything works like a charm

**Comment by PragTob** [comment]
Hi @arnebrasseur cooked up some build script maybe and hopefully this helps you! https://gist.github.com/arnebrasseur/4194d8c47e9493019e04

**Comment by Puspharaj** [comment]
Hi PragTob link is not working anymore (404). I do get the same error

pusp@home:~$ shoes/dist/shoes
internal:gem_prelude:1:in `require': cannot load such file -- rubygems.rb (LoadError)
    from <internal:gem_prelude>:1:in`<compiled>'
uninitialized constant Encoding::UTF_7

Someone please give me the solution to run shoes on Ubuntu 13.04 

Thanks in advance!!

**Comment by plexus** [comment]
Try https://gist.github.com/plexus/4194d8c47e9493019e04

**Comment by PragTob** [comment]
Ahh he changed his usernsame, so now this one here is the right one: https://gist.github.com/plexus/4194d8c47e9493019e04

**Comment by Puspharaj** [comment]
Hey Guys,

I just compiled it again by using ruby 1.9.2 in Ubuntu 13.04 now its working like charm! , I don't know what's the problem with Ruby 1.9.3 | Ruby 1.9.2 works. And Thanks again to quick Response from plexus and PragTob.

Here i mark the changes for what i have done to make it work. -- Hmm, here no option to highlight.

Install Depends
sudo apt-get install git-core libcairo2-dev libpixman-1-dev libpango1.0-dev libgif-dev libjpeg-dev libgtk2.0-dev vlc libvlc-dev portaudio19-dev libsqlite3-dev libcurl4-openssl-dev libxslt1-dev makeself curl libxslt-dev libxml2-dev libcurl4-openssl-dev libungif4-dev  ruby1.9.1-full -- This should be 1.9.2 (I hope).

pusp@home:~/shoes$ sudo apt-get install portaudio19-dev libjack-dev libjack0 will replace some other library - that's okay for me.

pusp@home:~/shoes$ sudo apt-get install ruby-rvm ---  didn't work so i used the below one.

http://askubuntu.com/questions/128966/how-do-i-install-ruby-rvm

curl -L get.rvm.io | bash -s stable
source ~/.rvm/scripts/rvm
rvm requirements
rvm install 1.9.2

#check this one
type rvm | head -1

##Install this
rvm install ruby-1.9.2
rvm use 1.9.2
gem install bundler

##Do this 
git clone git://github.com/shoes/shoes.git ${SHOES_DIR}
cd ${SHOES_DIR}
rvm use 1.9.2
bundle install
rake

Cheers!! let's grab some Beers!

**Comment by arka6** [comment]
Thanks Pushparaj! You saved my day, now is working on Debian wheezy 64.

**Comment by Puspharaj** [comment]
I'm glad that it worked for you. np!

**Comment by prostakov** [comment]
Had the same issue on Ubuntu 12.04 running ruby 1.9.3.
Compiled it with 1.9.2 works like a charm! Thank, you @Puspharaj!

**Comment by ccoupe** [comment]
Lets close this issue (months later). The problem is that when Ruby 1.9.3 starts up (initialized by Shoes) it looks for rbconfig.rb in the file system where the ruby was installed.  Since Shoes has a copy of Ruby at a different location all kinds of interesting error messages may occur.  The code in cache.rb that fixed things up barely worked with ruby 1.9.1 and 1.9.2 but it does not work "properly" with 1.9.3. If you convince your ruby to be built --enable-load-relative - there is a option to rvm to (re)build it that way (and you should) then you get a different set of error messages because Ruby 1.9.3 is HARDCODED to look for rubygems.rb  in lib/ruby/1.9.1 and Shoes has ruby/lib/   This all happens before the first line of Shoes.rb is run and there is nothing cache.rb can do about later.

I fixed both of those problems with my fork of Shoes. If those mostly harmless error messages (they are mostly harmless until you try to install gems inside Shoes) bothers you, then clone my fork and rebuild.

**Comment by Puspharaj** [comment]
Thanks for the resolution that you provided. Will check one more time..

Sent by using SONY mobile
